File: 02 - Even Fibonacci Numbers/python/main.py
def fiboEvenSum(limit):
    x = 1
    y = 2

    sum = y;

    while (y <= limit):
        buf = x + y
        x = y
        y = buf

        if (y % 2 == 0):
            sum += y

    return sum

result = fiboEvenSum(4000000)
print("result: " + str(result))

File: 02 - Even Fibonacci Numbers/java/src/Main.java
public class Main {
    public static void main(String[] args) {
        int result = fiboEvenSum(4000000);
        System.out.println("result: " + result);
    }

    static int fiboEvenSum(int limit) {
        int x = 1;
        int y = 2;
        int sum = y;
        while (y < limit) {
            int buf = x + y;
            x = y;
            y = buf;
            if (y % 2 == 0) {
                sum += y;
            }
        }
        return sum;
    }
}
